### PR TITLE
Fix empty paste

### LINKED ISSUE #2563: Pasting in Input with an empty string crashes the program
Pressing middle mouse button to paste causes this error in `Input._on_paste` if the clipboard is empty of text:

```
╭───────────────────────────────────────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────────────────────────────────────╮
│ /home/io/Projects/textual-paint/.venv/lib/python3.10/site-packages/textual/widgets/_input.py:335 in _on_paste                                                                       │
│                                                                                                                                                                                     │
│   332 │   │   │   event.prevent_default()                                                                                                                                           │
│   333 │                                                                                                                                                                             │
│   334 │   def _on_paste(self, event: events.Paste) -> None:                                                                                                                         │
│ ❱ 335 │   │   line = event.text.splitlines()[0]                                                                                                                                     │
│   336 │   │   self.insert_text_at_cursor(line)                                                                                                                                      │
│   337 │   │   event.stop()                                                                                                                                                          │
│   338                                                                                                                                                                               │
│                                                                                                                                                                                     │
│ ╭──────────────────────────────────────────────────────────── locals ────────────────────────────────────────────────────────────╮                                                  │
│ │ event = Paste(text='')                                                                                                         │                                                  │
│ │  self = CharInput(id='selected_color_char_input', classes={'color_well'}, pseudo_classes={'focus', 'focus-within', 'enabled'}) │                                                  │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                                                  │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
IndexError: list index out of range
```

To reproduce, use `printf "" | xsel` before pasting with MMB.

# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.22.3 |
| Rich    | 13.3.5 |

## Python

| Name           | Value                                            |
|----------------|--------------------------------------------------|
| Version        | 3.10.6                                           |
| Implementation | CPython                                          |
| Compiler       | GCC 11.3.0                                       |
| Executable     | /home/io/Projects/textual-paint/.venv/bin/python |

## Operating System

| Name    | Value                                                            |
|---------|------------------------------------------------------------------|
| System  | Linux                                                            |
| Release | 5.19.0-41-generic                                                |
| Version | #42~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 18 17:40:00 UTC 2 |

## Terminal

| Name                 | Value           |
|----------------------|-----------------|
| Terminal Application | vscode (1.77.3) |
| TERM                 | xterm-256color  |
| COLORTERM            | truecolor       |
| FORCE_COLOR          | *Not set*       |
| NO_COLOR             | *Not set*       |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=183, height=32 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 183                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 32                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |

### PR DESCRIPTION
**Pull request by rodrigogiraoserrao**
Closes #2563.

### CODE REVIEW

File: src/textual/widgets/_input.py [hunk 1]
```diff
@@ -332,7 +332,7 @@ async def _on_key(self, event: events.Key) -> None:
             event.prevent_default()
 
     def _on_paste(self, event: events.Paste) -> None:
-        line = event.text.splitlines()[0]
+        line = event.text.splitlines()[0] if event.text else ""
```
Comment (willmcgugan) [inline src/textual/widgets/_input.py:335]: Could we not skip the `insert_text_at_cursor` entirely when the pasted text is an empty string.

I'm assuming that inserting an empty string does nothing useful.

Comment (rodrigogiraoserrao) [reply src/textual/widgets/_input.py:335]: It doesn't do anything very useful, but that would short-circuit other things that might be expected, like an `Input.Changed` event.
Do you think it's worth adding that exception here?

Comment (willmcgugan) [reply src/textual/widgets/_input.py:335]: I don't think you would get a `Change` event, because nothing should have changed. Have you checked?

Comment (rodrigogiraoserrao) [reply src/textual/widgets/_input.py:335]: My bad, I thought `always_update` was `True` for reactives by default.

**Review by willmcgugan — APPROVED** [review]

**Pull request merged**